internal/server/middleware.go

```diff
@@ -182,3 +182,76 @@ func getCookie(req *http.Request, name string) (string, error) {
 	}
 	return url.QueryUnescape(cookie.Value)
 }
+
+func OrganizationFromDomain(defaultOrgName, defaultOrgDomain string) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		db := getDB(c)
+		host := c.Request.Header.Get("Host")
+
+		// remove port
+		if i := strings.Index(host, ":"); i >= 0 {
```

Comment:
This breaks on IPv6 addresses I suspect.

`net.SplitHostPort` seems close to what we want, but that seems to fail if there is no port. Maybe we could wrap it and ignore errors. Assuming an error means no port in the input string.